File: ragflow-automation-rearchitecture/selenium-testng-java/src/main/java/org/zemoso/pages/BasePage.java
package org.zemoso.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.zemoso.base.WebDriverFactory;

public abstract class BasePage {

    protected WebDriver driver;
    protected WebDriverWait wait;

    public BasePage(WebDriver driver) {
        this.driver = driver;
        wait = WebDriverFactory.getWait();
        PageFactory.initElements(driver, this);
    }
}

File: ragflow-automation-rearchitecture/selenium-csharp/Pages/BasePage.cs
using AutomationC_.Base;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;

namespace AutomationC_.Pages;

public abstract class BasePage
{
    protected IWebDriver Driver { get; }
    protected WebDriverWait Wait { get; }

    protected BasePage(IWebDriver driver)
    {
        Driver = driver;
        Wait = WebDriverFactory.Wait;
    }
}

